src/Core/Services/Implementations/SendService.cs

```diff
@@ -0,0 +1,170 @@
+                if (!org.MaxStorageGb.HasValue)
+                {
+                    throw new BadRequestException("This organization cannot use file sends.");
+                }
+
+                storageBytesRemaining = org.StorageBytesRemaining();
+            }
+
+            if (storageBytesRemaining < requestLength)
+            {
+                throw new BadRequestException("Not enough storage available.");
+            }
+
+            var fileId = Utilities.CoreHelpers.SecureRandomString(32, upper: false, special: false);
+            await _sendFileStorageService.UploadNewFileAsync(stream, send, fileId);
+
+            try
+            {
+                data.Id = fileId;
+                data.Size = stream.Length;
+                send.Data = JsonConvert.SerializeObject(data,
+                    new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
+                await SaveSendAsync(send);
+            }
+            catch
+            {
+                // Clean up since this is not transactional
+                await _sendFileStorageService.DeleteFileAsync(fileId);
+                throw;
+            }
+        }
+
+        public async Task DeleteSendAsync(Send send)
+        {
+            await _sendRepository.DeleteAsync(send);
+            if (send.Type == Enums.SendType.File)
+            {
+                var data = JsonConvert.DeserializeObject<SendFileData>(send.Data);
+                await _sendFileStorageService.DeleteFileAsync(data.Id);
+            }
+        }
+
+        // Response: Send, password required, password invalid
+        public async Task<(Send, bool, bool)> AccessAsync(Guid sendId, string password)
+        {
+            var send = await _sendRepository.GetByIdAsync(sendId);
+            var now = DateTime.UtcNow;
+            if (send == null || send.MaxAccessCount.GetValueOrDefault(int.MaxValue) <= send.AccessCount ||
+                send.ExpirationDate.GetValueOrDefault(DateTime.MaxValue) < now || send.Disabled ||
+                send.DeletionDate < now)
+            {
+                return (null, false, false);
+            }
+            if (!string.IsNullOrWhiteSpace(send.Password))
+            {
+                if (string.IsNullOrWhiteSpace(password))
+                {
+                    return (null, true, false);
+                }
+                var passwordResult = _passwordHasher.VerifyHashedPassword(new User(), send.Password, password);
+                if (passwordResult == PasswordVerificationResult.SuccessRehashNeeded)
+                {
+                    send.Password = HashPassword(password);
+                }
+                if (passwordResult == PasswordVerificationResult.Failed)
+                {
+                    return (null, false, true);
+                }
+            }
+            // TODO: maybe move this to a simple ++ sproc?
```

Comment:
I personally like atomic actions that do these kinds of incremental counters as close to the "source of truth" as possible (e.g. in the database), a `++` sproc might be a good option here for sure.